### PR TITLE
pythonPackages.opentracing: init at 2.2.0

### PR DESCRIPTION
**Pull request by Rakesh4G**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @CMCDragonkai

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -7019,4 +7019,9 @@
     githubId = 1986844;
     name = "Daniel Wheeler";
   };
+  rakesh4g = {
```
Comment (alexarice) [inline maintainers/maintainer-list.nix]: Can this be in a separate commit titled: `maintainers: add rakesh4g`

File: pkgs/development/python-modules/opentracing/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+buildPythonPackage rec {
+  pname = "opentracing";
+  version = "2.2.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "cfd231ba5c58f90bc277787e62861eb0c6e4af76e42957bec240bbdf71fb7e0e";
+  };
+
+#  preInstall = ''
```
Comment (alexarice) [inline pkgs/development/python-modules/opentracing/default.nix]: Why are these comments here?

File: pkgs/development/python-modules/opentracing/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+#    virtual-env env
+#    ../env/bin/activate
+#    make bootstrap
+#    make test
+#  '';
+
+  doCheck = false;
+
+  meta = with stdenv.lib; {
```
Comment (alexarice) [inline pkgs/development/python-modules/opentracing/default.nix]:
```suggestion
  meta = with lib; {
```
You can just use lib here if you put `lib` instead of `stdenv` in the function arguments at the top

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -6187,6 +6187,8 @@ in {
   runway-python = callPackage ../development/python-modules/runway-python { };
 
   pyprof2calltree = callPackage ../development/python-modules/pyprof2calltree { };
+
+  opentracing = callPackage ../development/python-modules/opentracing { };
```
Comment (alexarice) [inline pkgs/top-level/python-packages.nix]: I realise some may be in the wrong place but I believe this file should be in alphabetical order

Comment (multun) [reply pkgs/top-level/python-packages.nix]: it's so damn far from alphabetical order that's not useful anyway. I recently found something like 3 or 4 potential insertion points for some new package.

Comment (alexarice) [reply pkgs/top-level/python-packages.nix]: Fair enough, maybe this should be fixed at some point

Comment (jonringer) [reply pkgs/top-level/python-packages.nix]: some of the packages are meant to be clumped together based on relevance, however, the current state is very from either grouping or alphabetical sorting. I think we should just have a alphabetical sorted list given that we're now at 8,000+ python packages.

**Review by jonringer — COMMENTED** [review]
need futures for python2, and the tests were meant to be used with pytest. I would do something like:
```
{ lib, buildPythonPackage, fetchPypi, isPy27
, futures
, gevent
, mock
, pytest
, tornado
}:

buildPythonPackage rec {
  pname = "opentracing";
  version = "2.2.0";

  src = fetchPypi {
    inherit pname version;
    sha256 = "cfd231ba5c58f90bc277787e62861eb0c6e4af76e42957bec240bbdf71fb7e0e";
  };

  propagatedBuildInputs = lib.optional isPy27 futures;

  checkInputs = [ gevent mock pytest tornado ];
  checkPhase = ''
    pytest
  '';

  meta = with lib; {
    description = "This library is a Python platform API for OpenTracing";
    homepage = "https://github.com/opentracing/opentracing-python";
    license = licenses.asl20;
    maintainers = with maintainers; [ rakesh4g ];
  };
}
```

File: pkgs/development/python-modules/opentracing/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    homepage = https://github.com/opentracing/opentracing-python;
+    description = "This library is a Python platform API for OpenTracing";
```
Comment (jonringer) [inline pkgs/development/python-modules/opentracing/default.nix]: I would also shorten this, we already know it's a python library
```suggestion
    description = "Platform API for OpenTracing";
```

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (Rakesh4G) [reply maintainers/maintainer-list.nix]: ok @alexarice

File: pkgs/development/python-modules/opentracing/default.nix (hunk 2 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/opentracing/default.nix]: @alexarice I was trying to enable check phase as well. But was facing issues. That was the main reason for keeping the PR in draft.

File: pkgs/top-level/python-packages.nix (hunk 4 above)
Comment (Rakesh4G) [reply pkgs/top-level/python-packages.nix]: @alexarice , @multun , @jonringer Thanks for inputs. Yes, it is not very clear where to add new packages so I went for one at the end. I will try to put it back somewhere close to its group.

**Comment by Rakesh4G** [comment]
> { lib, buildPythonPackage, fetchPypi, isPy27
> , futures
> , gevent
> , mock
> , pytest
> , tornado
> }:
> 
> buildPythonPackage rec {
>   pname = "opentracing";
>   version = "2.2.0";
> 
>   src = fetchPypi {
>     inherit pname version;
>     sha256 = "cfd231ba5c58f90bc277787e62861eb0c6e4af76e42957bec240bbdf71fb7e0e";
>   };
> 
>   propagatedBuildInputs = lib.optional isPy27 futures;
> 
>   checkInputs = [ gevent mock pytest tornado ];
>   checkPhase = ''
>     pytest
>   '';
> 
>   meta = with lib; {
>     description = "This library is a Python platform API for OpenTracing";
>     homepage = "https://github.com/opentracing/opentracing-python";
>     license = licenses.asl20;
>     maintainers = with maintainers; [ rakesh4g ];
>   };
> }

Thanks @jonringer. Yes it looks more appropriate. I will test this in my environment and update.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (Rakesh4G) [reply maintainers/maintainer-list.nix]: https://github.com/NixOS/nixpkgs/pull/67817

File: pkgs/development/python-modules/opentracing/default.nix (hunk 5 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/opentracing/default.nix]: > I would also shorten this, we already know it's a python library

OK, I will update this with my next commit.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (Rakesh4G) [reply maintainers/maintainer-list.nix]: > Can this be in a separate commit titled: `maintainers: add rakesh4g`
OK @alexarice  , Now this is a separate commit.

File: pkgs/development/python-modules/opentracing/default.nix (hunk 3 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/opentracing/default.nix]: @alexarice Done.

File: pkgs/top-level/python-packages.nix (hunk 4 above)
Comment (Rakesh4G) [reply pkgs/top-level/python-packages.nix]: I have moved this this next to opencv4

File: pkgs/development/python-modules/opentracing/default.nix (hunk 5 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/opentracing/default.nix]: > I would also shorten this, we already know it's a python library
Done.

File: pkgs/development/python-modules/opentracing/default.nix (hunk 2 above)
Comment (Rakesh4G) [reply pkgs/development/python-modules/opentracing/default.nix]: > I will test this in my environment and update.

Done.

**Comment by Rakesh4G** [comment]
attaching the build_output along with test phase. 
[opentracing_build_with_tests.txt](https://github.com/NixOS/nixpkgs/files/3566009/opentracing_build_with_tests.txt)

**Comment by wd15** [comment]
I think you'll probably need to make the 7225cff commit (where you add yourself as maintainer) to be the first commit and then squash the rest into a second commit.

Build works for me though.

Result of `nix-review pr 67760` [1](https://github.com/Mic92/nix-review)
<details>
  <summary>2 packages were built:</summary>

  - python27Packages.opentracing
  - python37Packages.opentracing
</details>

**Comment by Rakesh4G** [comment]
> I think you'll probably need to make the [7225cff](https://github.com/NixOS/nixpkgs/commit/7225cffd9dbce16a487eca381c2b641385ca412d) commit (where you add yourself as maintainer) to be the first commit and then squash the rest into a second commit.
> 
> Build works for me though.
> 
> Result of `nix-review pr 67760` [1](https://github.com/Mic92/nix-review)
> 
> 2 packages were built:

Hi @wd15 , I have tried to update as suggested. Thanks.

**Comment by wd15** [comment]
On Wed, Sep 4, 2019 at 3:26 AM Rakesh Gupta <notifications@github.com> wrote:
>
> Hi @wd15 , I have tried to update as suggested. Thanks.

You could also [swap the
order](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) of
the two remaining commits.

-- 
Daniel Wheeler

**Comment by wd15** [comment]
> Hi @wd15 , I have tried to update as suggested. Thanks.

You could also [swap the order](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) of the two remaining commits.

**Comment by Rakesh4G** [comment]
> > Hi @wd15 , I have tried to update as suggested. Thanks.
> 
> You could also [swap the order](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) of the two remaining commits.

ok @wd15 , Thanks i will do this.

**Comment by Rakesh4G** [comment]
> > Hi @wd15 , I have tried to update as suggested. Thanks.
> 
> You could also [swap the order](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) of the two remaining commits.

Hi @wd15, When i went to reverse the commit order, I realised that the order in my branch is already in reverse order (add maintainer first).  But may be due to date difference it is not shown properly here.

```
> git log 
commit b871d8dd24451234f5db620873d3c4a40d00c7d3 (HEAD -> opentracing_python)
Author: Rakesh4G <rakeshgupta4u@gmail.com>
Date:   Fri Aug 30 20:44:50 2019 +1000

    python3Packages.opentracing: init at 2.2.0

commit 68fa9b6defa50ea19bfc16f0141a889a93c24fb0
Author: Rakesh4G <rakeshgupta4u@gmail.com>
Date:   Mon Sep 2 16:12:16 2019 +1000

    maintainers: add rakesh4g


```

**Comment by wd15** [comment]
> Hi @wd15, When i went to reverse the commit order, I realised that the order in my branch is already in reverse order (add maintainer first). But may be due to date difference it is not shown properly here.

No worries then.

**Comment by CMCDragonkai** [comment]
Is this ready to be merged?

**Comment by Rakesh4G** [comment]
@CMCDragonkai Yes, This is ready to be merged.

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
leaf package

**Comment by CMCDragonkai** [comment]
Hey can we get this merged?

**Comment by jonringer** [comment]
to fix the commit history, please do:

```
git rebase -i HEAD~3
# then delete the middle wrong commit, and switch the other two lines/commits
git pull -r origin master  # to rebase on top of current master
# resolve conflicts
git add .
git rebase --continue
```

**Comment by Rakesh4G** [comment]
> to fix the commit history, please do:
> 
> ```
> git rebase -i HEAD~3
> # then delete the middle wrong commit, and switch the other two lines/commits
> git pull -r origin master  # to rebase on top of current master
> # resolve conflicts
> git add .
> git rebase --continue
> ```

OK, Thanks.

**Comment by jonringer** [comment]
@GrahamcOfBorg build python27Packages.opentracing python37Packages.opentracing python38Packages.opentracing

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
commits LGTM
outputs LGTM

```
[5 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/67760
3 package were build:
python27Packages.opentracing python37Packages.opentracing python38Packages.opentracing
```
